packages/documentation/.storybook/addons/styles-switcher/StylesSwitcher.tsx

```diff
@@ -94,6 +96,49 @@ function StylesSwitcher() {
     );
   }, [preview, currentTheme, currentChannel]);
 
+  /**
+   * Sets the design system styles import SCSS file to the correct theme and channel file
+   */
+  useEffect(() => {
+    if (!stylesCodeBlocks) return;
+
+    const t = currentTheme.toLowerCase();
+    const c = currentChannel.toLowerCase();
+    const domain = "'@swisspost/design-system-styles/";
+
+    stylesCodeBlocks.forEach(stylesCodeBlock => {
+      let sourceArray = stylesCodeBlock.querySelectorAll('.token.string');
+      if (sourceArray.length) {
+        sourceArray.forEach((s, i) => {
+          let source = s.innerHTML;
+          // Remove the domain from the source to make sure we don't override it
+          source = source.replace(domain, '');
+
+          // Check if one of the themes or channels are in the scss path
+          let theme = THEMES.find(t => source.indexOf(t.toLowerCase()) > -1);
+          let channel = CHANNELS.find(c => source.indexOf(c.toLowerCase()) > -1);
+
+          const updateTheme = theme && theme.toLowerCase() !== t;
+          const updateChannel = channel && channel.toLowerCase() !== c;
```

Comment:
Would that work too?
```suggestion
          const updateTheme = theme?.toLowerCase() !== t;
          const updateChannel = channel?.toLowerCase() !== c;
```

Reply:
no, that does not work because it tries to apply toLowerCase() to undefined so it breaks